components/bpmn/bpmn-rest/src/main/java/org/wso2/carbon/bpmn/rest/service/stats/ProcessAndTaskService.java

```diff
@@ -0,0 +1,482 @@
+        long countOfFailedTaskInstances = BPMNOSGIService.getManagementService()
+                .createJobQuery().jobTenantId(str).withException().count();
+
+        if(countOfCompletedTaskInstances == 0 && countOfActiveTaskInstances == 0 &&
+                countOfSuspendedTaskInstances == 0 && countOfFailedTaskInstances == 0){
+            response.setData(taskCountList);
+        }
+        else {
+            completedTaskInstances = new ProcessTaskCount();
+            completedTaskInstances.setStatusOfProcessOrTask("Completed");
+            completedTaskInstances.setCount(countOfCompletedTaskInstances);
+            taskCountList.add(completedTaskInstances);
+
+            activeTaskInstances = new ProcessTaskCount();
+            activeTaskInstances.setStatusOfProcessOrTask("Active");
+            activeTaskInstances.setCount(countOfActiveTaskInstances);
+            taskCountList.add(activeTaskInstances);
+
+            suspendedTaskInstances = new ProcessTaskCount();
+            suspendedTaskInstances.setStatusOfProcessOrTask("Suspended");
+            suspendedTaskInstances.setCount(countOfSuspendedTaskInstances);
+            taskCountList.add(suspendedTaskInstances);
+
+            failedTaskInstances = new ProcessTaskCount();
+            failedTaskInstances.setStatusOfProcessOrTask("Failed");
+            failedTaskInstances.setCount(countOfFailedTaskInstances);
+            taskCountList.add(failedTaskInstances);
+
+            response.setData(taskCountList);
+        }
+        return response;
+    }
+
+    /**
+     * Get the average time duration of completed processes
+     *
+     * @return list with the completed processes and the average time duration taken for each process
+     */
+    @GET
+    @Path("/avgDurationToCompleteProcess/")
+    @Produces({MediaType.APPLICATION_JSON, MediaType.APPLICATION_XML})
+    public ResponseHolder getAvgTimeDurationForCompletedProcesses() {
+        List<ProcessDefinition> deployements = BPMNOSGIService.getRepositoryService().
+                createProcessDefinitionQuery().processDefinitionTenantId(str).list();
+
+        ResponseHolder response = new ResponseHolder();
+        List list = new ArrayList<>();
+
+        for (ProcessDefinition instance : deployements) {
+            CompletedProcesses bpmnProcessInstance = new CompletedProcesses();
+            bpmnProcessInstance.setProcessDefinitionId(instance.getId());
+
+            double totalTime = 0;
+            double averageTime = 0;
+            String processDefinitionID = instance.getId();
+
+
+            HistoricProcessInstanceQuery historicProcessInstanceQuery = BPMNOSGIService.
+                    getHistoryService().createHistoricProcessInstanceQuery().
+                    processInstanceTenantId(str).
+                    processDefinitionId(processDefinitionID).finished();
+
+            long noOfHistoricInstances = historicProcessInstanceQuery.count();
+
+            if (noOfHistoricInstances == 0) {
```

Comment:
Change this to logic within the if condition. Always implement logic within if and the compliment in else part and do not leave empty if {}